lib/msf/core/exploit/remote/http_client.rb

```diff
@@ -235,7 +235,8 @@ def connect(opts={})
   # @raise [Rex::Proto::Http::WebSocket::WebSocketError] raises an exception if the connection fails
   # @return [Rex::Proto::Http::WebSocket::Interface]
   def connect_ws(opts={}, timeout = 20)
-    ws_key = Rex::Text.rand_text_alphanumeric(20)
+    # As per the spec (RFC6455 Section 11.3.1), a Sec-WebSocket-Key is a 16 byte value that has been Base64 encoded.
+    ws_key = Rex::Text.encode_base64(Rex::Text.rand_text_alphanumeric(16))
```

Comment:
Not a blocker

```suggestion
    ws_key = Rex::Text.encode_base64(SecureRandom.bytes(16))
```

(I imagine SecureRandom is already `require`'d in this context)

Reply:
yup tested and `SecureRandom.bytes` works great, and is a better choice over `Rex::Text.rand_text_alphanumeric`, thanks Alan!